tests/services/milestone.service.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { of } from 'rxjs';
+import { Milestone } from '../../src/app/core/models/milestone.model';
+import { GithubService } from '../../src/app/core/services/github.service';
+import { MilestoneService } from '../../src/app/core/services/milestone.service';
+
+let milestoneService: MilestoneService;
+let githubServiceSpy: jasmine.SpyObj<GithubService>;
+
+describe('MilestoneService', () => {
+  beforeEach(() => {
+    githubServiceSpy = jasmine.createSpyObj('GithubService', ['fetchAllMilestones']);
+    milestoneService = new MilestoneService(githubServiceSpy);
+  });
+
+  describe('MilestoneService: fetchMilestones()', () => {
+    it('should fetch all milestones', async () => {
+      const mockMilestones = [{ title: 'Milestone 1' }, { title: 'Milestone 2' }];
+      githubServiceSpy.fetchAllMilestones.and.returnValue(of(mockMilestones));
+      milestoneService.fetchMilestones().subscribe((response) => {
+        expect(githubServiceSpy.fetchAllMilestones).toHaveBeenCalled();
+        expect(response).toEqual(mockMilestones);
+        expect(milestoneService.milestones.length).toBe(3);
+        expect(milestoneService.milestones[0].title).toBe('Milestone 1');
+        expect(milestoneService.hasNoMilestones).toBeFalse();
+      });
```

Comment:
Consider using Jasmine's `done()` callback at the end of the assertions. This is to ensure that the test doesn't complete before the value is emitted, which would lead to the assertions not being executed by Jasmine.

The same issue applies to the test case below.

Reply:
I looked into this, and received an Error Log.

<img width="1089" alt="Screenshot 2024-03-15 at 2 11 33 PM" src="https://github.com/CATcher-org/WATcher/assets/114080910/663a2f08-95e4-427d-a284-987f2711c1cc">

Reply:
If I am not wrong, async/await cannot be used with done. You try remove the async keyword.